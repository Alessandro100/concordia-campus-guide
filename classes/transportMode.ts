enum transportMode {
    driving = 'driving',
    walking = 'walking',
    bicycle = 'bicycle',
    transit = 'transit'
}

export default transportMode;