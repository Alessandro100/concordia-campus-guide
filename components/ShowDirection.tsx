import React, { Component } from 'react';
import { View, StyleSheet, Text, Dimensions } from "react-native";
import Polyline from '@mapbox/polyline'
import MapView, { PROVIDER_GOOGLE } from 'react-native-maps';


class ShowDirection extends Component {

  constructor(props) {
super(props)
  
    this.state = {
      coords:[],
      // Region is set to fit Concordia campus.
      region: {
        latitude: 45.4582,
        longitude: -73.6405,
        desLatitude: 45.497406,
        desLongitude: -73.577102,
        latitudeDelta: 0,
        longitudeDelta: 0.1,
      },
      // Hard coded adresses to display path.
      startLocation: "7141 Rue Sherbrooke Ouest, Montréal, QC H4B 1R6",
      destination: "1455 Boulevard de Maisonneuve O, Montréal, QC H3G 1M8",
      x: 'true',

      
    }
    this.getDirections = this.getDirections.bind(this);
  }

    // This function takes adresses as input and call the google API.
    // Time and distance are unused but left here for potential extra.
    async getDirections(startLoc, desLoc) {
     
        try {
          const resp = await fetch(`https://maps.googleapis.com/maps/api/directions/json?origin=${startLoc}&destination=${desLoc}&mode=transit&key=AIzaSyDpY_ACPWoo3mVPPmiLKQe1aBhjkDYlwJI`);
          const respJson = await resp.json();
          const response = respJson.routes[0]
          const distanceTime = response.legs[0];
          const distance = distanceTime.distance.text;
          const time = distanceTime.duration.text;
          const points = Polyline.decode(respJson.routes[0].overview_polyline.points);
          let coords = points.map((point, index) => {
            return  {
                latitude : point[0],
                longitude : point[1]
            }
        })
        console.log(coords, 'coords');
        this.setState({coords: coords});
        return coords;
        } catch(error) {
        
        }
      }

     componentDidMount() {
       this.getDirections(this.state.startLocation, this.state.destination);
     } 

    render(){
        
        return( 
        <MapView 
            showsUserLocation
            style={styles.mapStyle}
            initialRegion={this.state.region}
          >
      
          <MapView.Polyline
            strokeWidth={2}
            strokeColor="red"
            coordinates={this.state.coords}
          />
        </MapView>
        )
    }
}


const styles = StyleSheet.create({
  
  mapStyle: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,

 },
});

export default ShowDirection;