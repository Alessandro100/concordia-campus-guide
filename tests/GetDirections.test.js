import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import ShowDirection from '../components/ShowDirection';
import transportMode from '../classes/transportMode';
import MapView, { Marker, Polyline } from 'react-native-maps';
import Location from '../classes/location';
import OutdoorPOI from '../classes/outdoorPOI';

describe('Get Directions Suite', () => {
  test('ShowDirection snapshot', () => {
    const tree = renderer
      .create(<ShowDirection />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Map Elements Loaded', async (done) => {
    const wrapper = mount(
      <ShowDirection
        startLocation={new OutdoorPOI(new Location(45.458488, -73.639862), 'test-start')}
        endLocation={new OutdoorPOI(new Location(45.50349, -73.572182), 'test-end')}
        transportType={transportMode.transit}
      />
    );
    const instance = wrapper.instance();
    try {
      await instance.loadRoute();
      expect(wrapper.find(Marker)).toHaveLength(2);
      expect(wrapper.getElements(<MapView.Polyline />).length > 0).toEqual(true);
    } catch (e) {
      done();
    }
  });
  test('Pin Location', done => {
    const wrapper = mount(
      <ShowDirection
        startLat={45.458488}
        startLon={-73.6398621}
        endLat={45.50349}
        endLon={-73.572182}
        transportType={transportMode.transit}
      />
    );
    const instance = wrapper.instance();
    const startingLocation = new Location(45.458488, -73.6398621);
    const startingPOI = new OutdoorPOI(startingLocation, 'startingLocation');
    expect(instance.getPinLocation(startingPOI.getLocation())).toEqual({ latitude: 45.458488, longitude: -73.6398621 });
    expect(instance.getPinLocation(null)).toEqual({ latitude: 0, longitude: 0 });
    done();
  });
});
