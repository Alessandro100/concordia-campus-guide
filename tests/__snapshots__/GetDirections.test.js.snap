// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Directions Suite ShowDirection snapshot 1`] = `null`;
